packages/base/src/asset-registries/Icons.js

```diff
@@ -1,19 +1,26 @@
 import { registerIcon, registerCollectionPromise } from "../SVGIconRegistry.js";
-import { fetchJsonOnce } from "../util/FetchHelper.js";
-import { getEffectiveAssetPath } from "../util/EffectiveAssetPath.js";
 
+/**
+ * @deprecated
+ */
 const registerIconBundle = async (collectionName, bundleData) => {
+	throw new Error("This method has been removed. Use `registerIconLoader` instead.");
+};
+
+const registerIconLoader = async (collectionName, loader) => {
 	let resolveFn;
 	const collectionFetched = new Promise(resolve => {
 		resolveFn = resolve;
 	});
 	registerCollectionPromise(collectionName, collectionFetched);
 
-	if (typeof bundleData !== "object") { // not inlined from build -> fetch it
-		bundleData = await fetchJsonOnce(getEffectiveAssetPath(bundleData));
+	try {
+		const iconData = await loader();
```

Comment:
I still think this `register*` function should not behave differently than the rest

I18n:
```
const registerI18nLoader = (packageName, localeId, loader) => {
	// register loader by key
	const bundleKey = `${packageName}/${localeId}`;
	loaders.set(bundleKey, loader);
};
```

Themes:
```
const registerThemePropertiesLoader = (packageName, themeName, loader) => {
	loaders.set(`${packageName}/${themeName}`, loader);
	registeredPackages.add(packageName);
	registeredThemes.add(themeName);
};
```

Locale Data:
```
const registerLocaleDataLoader = (localeId, loader) => {
	loaders.set(localeId, loader);
};
```

and Icons is quite different:
```
const registerIconLoader = async (collectionName, loader) => {
	let resolveFn;
	const collectionFetched = new Promise(resolve => {
		resolveFn = resolve;
	});
	registerCollectionPromise(collectionName, collectionFetched);

	try {
		const iconData = await loader();
		fillRegistry(iconData);
		resolveFn();
	} catch (e) {
		console.error(e.message); /* eslint-disable-line */
	}
};
```

IMO we should split it into a "true" register function, which will store the loader in a set or map, and then a "fetch" function which will explicitly call it. Then if we want to be compatible, we will call the "fetch" function in the assets file:

For example something like:

```
import { registerIconLoader, fetchIcons } from "@ui5/webcomponents-base/dist/asset-registries/Icons.js";

const loadIconsBundle = async () => {
    const iconData = (await import("../generated/assets/SAP-icons.json")).default;
    if (typeof iconData === "string" && iconData.endsWith(".json")) {
        throw new Error("[icons] Invalid bundling detected - dynamic JSON imports bundled as URLs. Switch to inlining JSON files from the build or use `import \"@ui5/webcomponents-icons/dist/Assets-static.js\". Check the \"Assets\" documentation for more information.");
    }
    return iconData;
}

registerIconLoader("SAP-icons", loadIconsBundle);
fetchIcons("SAP-icons");
```

Reply:
refactored: `SVGIconRegistry.js` moved to `asset-registries/Icons.js`
registerIconLoader no longer fetches
getIconData (async) fetches and fills the registry the first time